### PR TITLE
[DROOLS-6483] Implement TextIndex

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @kostola  @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-6483

Need https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1713

To be viewed/merged after #3720

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-commons/pom.xml [hunk 1]
```diff
@@ -37,5 +46,14 @@
     </dependency>
   </dependencies>
 
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-text</artifactId>
+        <version>${version.common-text}</version>
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-commons/pom.xml]: I would suggest to move this dependency management somewhere to kie parent.

File: kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/model/expressions/KiePMMLTextIndexNormalizationTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        KiePMMLInlineTable inlineTable = new KiePMMLInlineTable("inlineTable", Collections.emptyList(), Collections.singletonList(row));
+        KiePMMLTextIndexNormalization indexNormalization = KiePMMLTextIndexNormalization.builder("indexNormalization", Collections.emptyList())
+                .withKiePMMLInlineTable(inlineTable)
+                .withIsCaseSensitive(false)
+                .build();
+        String text = "interfacea";
+        String retrieved = indexNormalization.replace(text, true, 0, false, DEFAULT_TOKENIZER);
+        assertEquals("fooa", retrieved);
+
+        //---
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/model/expressions/KiePMMLTextIndexNormalizationTest.java:47]: Is this comment intentional?

**Review by jiripetrlik — COMMENTED** [review]
I have some comments.

File: kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/model/expressions/KiePMMLTextIndexNormalizationTest.java (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/model/expressions/KiePMMLTextIndexNormalizationTest.java:47]: @jiripetrlik 
Yes - the test is actually run with two different inputs - I use the comment to clearly separate them

**Review by danielezonca — APPROVED** [review]
I agree with Jiri about moving dependency declaration to kie-parent, once solved this approved 👍

File: kie-pmml-trusty/kie-pmml-commons/pom.xml (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-commons/pom.xml]: @jiripetrlik @danielezonca 

https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1713

**Pull request merged**